### DISCUSSION TITLE
Warping even sized matrices in the nearest mode causes 0 to apear in the last row/column

### DISCUSSION THREAD

**Discussion by magnux** [kornia/kornia#865] [Questions] [answered]
## 🐛 Bug

Warping even sized matrices in the nearest mode causes 0 to apear in the last row/column.
The odd sized matrices are handled correctly, so it seems some rounding or border case issue.

## To Reproduce
```
import torch
import kornia

torch.manual_seed(0)
M = torch.eye(3).unsqueeze(0)

dsize = (3, 3)
src = torch.ones(1, 3, *dsize)
warper = kornia.HomographyWarper(*dsize, mode='nearest')
warper.precompute_warp_grid(M)

print(warper(src))

dsize = (4, 4)
src = torch.ones(1, 3, *dsize)
warper = kornia.HomographyWarper(*dsize, mode='nearest')
warper.precompute_warp_grid(M)

print(warper(src))
```
Results:

```
tensor([[[[1., 1., 1.],
          [1., 1., 1.],
          [1., 1., 1.]],

         [[1., 1., 1.],
          [1., 1., 1.],
          [1., 1., 1.]],

         [[1., 1., 1.],
          [1., 1., 1.],
          [1., 1., 1.]]]])

tensor([[[[1., 1., 1., 0.],
          [1., 1., 1., 0.],
          [1., 1., 1., 0.],
          [0., 0., 0., 0.]],

         [[1., 1., 1., 0.],
          [1., 1., 1., 0.],
          [1., 1., 1., 0.],
          [0., 0., 0., 0.]],

         [[1., 1., 1., 0.],
          [1., 1., 1., 0.],
          [1., 1., 1., 0.],
          [0., 0., 0., 0.]]]])
```

## Expected behavior

Results:

```
tensor([[[[1., 1., 1.],
          [1., 1., 1.],
          [1., 1., 1.]],

         [[1., 1., 1.],
          [1., 1., 1.],
          [1., 1., 1.]],

         [[1., 1., 1.],
          [1., 1., 1.],
          [1., 1., 1.]]]])

tensor([[[[1., 1., 1., 1.],
          [1., 1., 1., 1.],
          [1., 1., 1., 1.],
          [1., 1., 1., 1.]],

         [[1., 1., 1., 1.],
          [1., 1., 1., 1.],
          [1., 1., 1., 1.],
          [1., 1., 1., 1.]],

         [[1., 1., 1., 1.],
          [1., 1., 1., 1.],
          [1., 1., 1., 1.],
          [1., 1., 1., 1.]]]])
```

## Environment

kornia 0.4.0

**Comment by edgarriba** [comment, marked as answer]
hi @magnux , thanks for reporting this issue.

I believe that this is related to https://github.com/kornia/kornia/issues/747 that refers to the `align_corners` flag for the `grid_sample` function. Check my comment here: https://github.com/kornia/kornia/issues/747#issuecomment-716763903

Ideally we should find a proper configuration for aligning corners. /cc @pmeier was trying to normalize this across different functions too.

**Comment by magnux** [comment]
Calling HomographyWarper with align_corners=True indeed solves the issue.
It seems related with #747
